pandas/core/resample.py

```diff
@@ -1357,20 +1365,51 @@ def __init__(
         self.convention = convention or "E"
         self.convention = self.convention.lower()
 
-        if isinstance(loffset, str):
-            loffset = to_offset(loffset)
-        self.loffset = loffset
-
         self.how = how
         self.fill_method = fill_method
         self.limit = limit
-        self.base = base
+
+        self.origin = Timestamp(origin) if origin is not None else None
+        self.offset = Timedelta(offset) if offset is not None else None
 
         # always sort time groupers
         kwargs["sort"] = True
 
+        self._warn_if_loffset_or_base_is_used(base, loffset, _warning_stack_level)
+        if isinstance(loffset, str):
+            loffset = to_offset(loffset)
+        self.loffset = loffset
+        self.base = base if base else 0
+
         super().__init__(freq=freq, axis=axis, **kwargs)
 
+    @staticmethod
+    def _warn_if_loffset_or_base_is_used(base, loffset, _warning_stack_level):
+        if base is not None:
+            warnings.warn(
+                "'base' in .resample() and in Grouper() is deprecated.\n"
+                "The new arguments that you should use "
+                "are 'offset' or 'origin'.\n\n"
+                '>>> df.resample(freq="3s", base=2)\n'
```

Comment:
is there a case someone using base would actually need origin?

Reply:
Well, I thought that some people may use the base argument when they really want to use 'origin' instead. I mean, I wanted to use base with some nasty hack, but I'm doing this PR instead :sweat_smile:. But maybe that should not be in this warning message :wink: 

Let me know if you want me to remove the mention of this argument here!